subsys/bluetooth/audio/tmap.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/types.h>
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include "audio_internal.h"
+#include <zephyr/bluetooth/audio/tmap.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bt_tmap, CONFIG_BT_TMAP_LOG_LEVEL);
+
+static uint16_t tmap_role;
+static struct bt_tmap_cb *cb;
+static bool tmas_found;
+
+static struct bt_uuid_16 uuid = BT_UUID_INIT_16(0);
+static struct bt_gatt_discover_params discover_params;
+static struct bt_gatt_read_params read_params;
```

Comment:
If we want to support multiple discovery procedures in parallel (i.e. for multiple connections), then these values need to be in an array of some sort.